File: app/src/main/java/com/mohnage7/movies/MoviesApplication.java
package com.mohnage7.movies;

import android.app.Application;

import com.mohnage7.movies.di.component.DaggerDataComponent;
import com.mohnage7.movies.di.component.DataComponent;
import com.mohnage7.movies.di.module.DataModule;

public class MoviesApplication extends Application {

    private static MoviesApplication instance;
    DataComponent dataComponent;

    public static MoviesApplication getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        initDataComponent();
    }

    private void initDataComponent() {
        dataComponent = DaggerDataComponent.builder().
                dataModule(new DataModule(this))
                .build();
        dataComponent.inject(this);
    }

    public DataComponent getDataComponent() {
        return dataComponent;
    }
}

File: app/src/main/java/com/mohnage7/movies/MoviesApplication.kt
package com.mohnage7.movies

import android.app.Application
import com.mohnage7.movies.db.di.dataBaseModule
import com.mohnage7.movies.features.moviedetails.di.movieDetailsModule
import com.mohnage7.movies.features.movies.di.moviesModule
import com.mohnage7.movies.network.di.networkModule
import org.koin.android.ext.koin.androidContext
import org.koin.core.context.startKoin


class MoviesApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        startKoinInjection();
    }

    private fun startKoinInjection() {
        startKoin {
            androidContext(this@MoviesApplication)
            modules(listOf(networkModule, dataBaseModule, moviesModule, movieDetailsModule))
        }
    }
}
